### DISCUSSION TITLE
Add .root modifier to x-intersect

### DISCUSSION THREAD

**Discussion by digitall-it** [alpinejs/alpine#3292] [7. Bugs (archived)]
### Discussed in https://github.com/alpinejs/alpine/discussions/2993

<div type='discussions-op-text'>

<sup>Originally posted by **digitall-it** June 16, 2022</sup>
If x-intersect is used on a sticky header table to detect when a cell comes into view, it may lead to unexpected result, for the fact that the IntersectionObserver seems to be always initialized to check for intersection related to the page element, and there doesn't seem to be a way to specify the root. I would like to implement it into a Livewire component so I had to resort to putting the observer initializer into x-data.</div>

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/alpinejs/alpine/discussions/1423
